### PR TITLE
PDA Messages are once again shown to non-observing dead mobs

### LINKED ISSUE #77748: PDA Messages are now only shown to "dead" players and not observers/admins
Reporting client version: 514.1589

## Round ID:
[212868](https://scrubby.melonmesa.com/round/212868)

## Testmerges:
- [Trying to qdelete a callback now lags the server instead.](https://www.github.com/tgstation/tgstation/pull/77447)
- [MODLink System (+ NWTLMM)](https://www.github.com/tgstation/tgstation/pull/77639)

## Reproduction:

Seems related to https://github.com/tgstation/tgstation/pull/77590

Possibly a better method to reproduce, but requires 3 people this way...
Observe round and have two people spawn in with PDAs and send messages. > You'll see the notification "noise" for the message but not the message as an observer.
Spawn yourself in and delete yourself so you now show as "Dead"
Have them send messages again and they now show just fine

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

Current implementation uses isobserver for checking if we should send to the player but dead players (who are still in their body) are not observers and will fail that check.

## Why It's Good For The Game
Fixes https://github.com/tgstation/tgstation/issues/77748

## Changelog

:cl:
fix: pda messages are no longer hidden from dead players who are not observers
/:cl:

### CODE REVIEW

**Comment by VladinXXV** [comment]
isn't the problem that it was ONLY being shown to non-observing dead mobs, not observers or admins? title seems a bit backwards here.

**Comment by MrMelbert** [comment]
Please fill out your PR template, your PR does not A. explain what is causing the bug, B. explain how it is fixing the bug, or C. include a changelog entry for a playing facing bug

File: code/modules/modular_computers/file_system/programs/messenger/messenger_program.dm [hunk 1]
```diff
@@ -608,9 +608,24 @@
...
+
+		for(var/mob/player_mob as anything in to_process)
+			if(isnull(player_mob.client))
+				if(!did_scream_about_null_client)
+					stack_trace("[listener_key] contained a mob with a null client.")
+					did_scream_about_null_client = TRUE
+				to_process -= player_mob
+				continue
+
+			if(!(player_mob.client.prefs.chat_toggles & CHAT_GHOSTPDA))
```
Comment (MrMelbert) [inline code/modules/modular_computers/file_system/programs/messenger/messenger_program.dm:627]: Why are we not using `get_chat_toggles`?

File: code/modules/modular_computers/file_system/programs/messenger/messenger_program.dm [hunk 2]
```diff
@@ -608,9 +608,24 @@
 
 	// Show it to ghosts
 	var/ghost_message = span_game_say("[span_name("[sender]")] [rigged ? "(as [span_name(fake_name)]) Rigged " : ""]PDA Message --> [span_name("[signal.format_target()]")]: \"[signal.format_message()]\"")
-	var/list/ghosts = GLOB.dead_player_list + GLOB.current_observers_list
-	for(var/mob/player_mob as anything in ghosts)
-		if(isobserver(player_mob) && (get_chat_toggles(player_mob.client) & CHAT_GHOSTPDA))
+	var/list/message_listeners = list(
+		"dead players list" = GLOB.dead_player_list,
+		"observing players list" = GLOB.current_observers_list,
+	)
```
Comment (MrMelbert) [inline code/modules/modular_computers/file_system/programs/messenger/messenger_program.dm:614]: This seems over-engineered for what is going on here

Comment (ZephyrTFA) [reply code/modules/modular_computers/file_system/programs/messenger/messenger_program.dm:614]: Doing it this way to identify null clients in the list w/o having to say null in one of these two lists which is not helpful

**Pull request merged**